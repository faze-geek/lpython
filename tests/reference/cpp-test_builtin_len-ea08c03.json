{
    "basename": "cpp-test_builtin_len-ea08c03",
    "cmd": "lpython --no-color --show-cpp {infile}",
    "infile": "tests/../integration_tests/test_builtin_len.py",
    "infile_hash": "39ba468d5d0b0ed6fe1bf20a158ba9448e9b799f0277d994588265cd",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "cpp-test_builtin_len-ea08c03.stdout",
    "stdout_hash": "8cf1c47cc1e3b0a11a6497763cba7231dffbfe416c843251bc727b97",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}