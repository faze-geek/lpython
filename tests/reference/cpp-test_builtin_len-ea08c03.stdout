#include <iostream>
#include <string>
#include <vector>
#include <cassert>
#include <cmath>
#include <complex>
#include <Kokkos_Core.hpp>
#include <lfortran_intrinsics.h>

template <typename T>
Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
{
    Kokkos::View<T*> r("r", v.size());
    for (size_t i=0; i < v.size(); i++) {
        r(i) = v[i];
    }
    return r;
}

void _lpython_main_program()
{
    test_len();
}

void test_len()
{
    int i;
    std::string s;
    std::string t;
    s = "abcd";
    assert (s.size() == 4);
    s = "";
    assert (s.size() == 0);
    assert (std::string("abcd").size() == 4);
    assert (std::string("").size() == 0);
    t = "efg";
    s = std::string("abcd") + std::string(t);
    i = s.size();
    assert (i == 7);
    i = std::string(std::string("abc") + std::string("def")).size();
    assert (i == 6);
}

namespace {

void main2() {
        _lpython_main_program();
}

}
int main(int argc, char* argv[])
{
    Kokkos::initialize(argc, argv);
    main2();
    Kokkos::finalize();
    return 0;
}
